build/src/lib.rs

```diff
@@ -148,52 +281,59 @@ pub struct BinaryBuilder {
     fallback_to_build_dir: bool,
     build_dir: Option<PathBuf>,
     outputs: Vec<PathBuf>,
+    transform_path_fn: Option<Box<dyn TransformPathFn>>,
+    transform_data_fn: Option<Box<dyn TransformDataFn<Cursor<Vec<u8>>>>>,
 }
 
 impl BinaryBuilder {
+    /// ```plain
```

Comment:
Some docstrings need cleanup but let's look at that once the code changes are done.